--- a/src/SDIO/driver_fw/driver/aic8800/aic8800_bsp/aic_bsp_driver.h	2025-01-03 18:56:39.057562241 +0900
+++ b/src/SDIO/driver_fw/driver/aic8800/aic8800_bsp/aic_bsp_driver.h	2025-01-03 18:56:51.005688576 +0900
@@ -502,7 +502,7 @@
 
 
 #define FEATURE_SDIO_CLOCK          50000000 // 0: default, other: target clock rate
-#define FEATURE_SDIO_CLOCK_V3       150000000 // 0: default, other: target clock rate
+#define FEATURE_SDIO_CLOCK_V3       50000000 // 0: default, other: target clock rate
 #define FEATURE_SDIO_PHASE          2        // 0: default, 2: 180°
 
 struct aicbt_patch_table {
